unleash-yggdrasil/src/strategy_parsing.rs

```diff
@@ -327,6 +330,30 @@ fn rollout_constraint(mut node: Pairs<Rule>) -> RuleFragment {
     })
 }
 
+fn get_hostname() -> Result<String, SdkError> {
+    //This is primarily for testing purposes
+    if let Ok(hostname_env) = env::var("HOSTNAME") {
+        return Ok(hostname_env);
+    }
+
+    hostname::get()
+        .map_err(|_| SdkError::StrategyEvaluationError)
+        .and_then(|os_str| {
+            os_str
+                .into_string()
+                .map_err(|_| SdkError::StrategyEvaluationError)
+        })
+}
+
+fn hostname_constraint(inverted: bool, mut node: Pairs<Rule>) -> RuleFragment {
+    let target_hostnames = harvest_set(node.next().unwrap().into_inner());
+
+    Box::new(move |_: &Context| match get_hostname() {
+        Ok(hostname) => target_hostnames.contains(&hostname).invert(inverted),
+        Err(_) => false,
```

Comment:
I've had to make a decision here because I don't see any other SDK outside of Rust handling this error case (this is an OS call or env var read so this _can_ fail). This would disable the strategy if retrieving the hostname fails, means means the strategy would be ignored